ReleaseNotes: add lld/ELF notes

MaskRay opened the pull request.

MaskRay committed:
ReleaseNotes: add lld/ELF notes

Pull Request: https://github.com/llvm/llvm-project/pull/124508

smithp35 reviewed (commented):
Spotted a small typo, but otherwise LGTM. I'll have a check to see if there's anything else I can suggest.

smithp35 commented:
Suggestions for addition:
Assuming it has made the release, this is a potentially breaking change for some people.
*  https://github.com/llvm/llvm-project/pull/124068 Error for executable .note.GNU-stack unless -z execstack or -r
* [LLD] Deprecate --lto-basic-block-sections=labels (#110697) This option is now replaced by `--lto-basic-block-address-map`.

MaskRay commented on lld/docs/ReleaseNotes.rst:
```diff
@@ -29,12 +29,35 @@ ELF Improvements
   globally changing the semantics of section matching. It also independently
   increases the expressive power of linker scripts.
   (`#95323 <https://github.com/llvm/llvm-project/pull/95323>`_)
+* The ``achivename:`` syntax when matching input sections is now supported.
```
Thanks! Fixed

MaskRay commented:
> Suggestions for addition: Assuming it has made the release, this is a potentially breaking change for some people.
> 
> * [[ELF] Error for executable .note.GNU-stack unless -z execstack or -r #124068](https://github.com/llvm/llvm-project/pull/124068) Error for executable .note.GNU-stack unless -z execstack or -r
> 
> Not strictly a breaking change, but worth highlighting.
> 
> * [LLD] Deprecate --lto-basic-block-sections=labels ([[LLD] Deprecate --lto-basic-block-sections=labels #110697](https://github.com/llvm/llvm-project/pull/110697)) This option is now replaced by `--lto-basic-block-address-map`.

Thanks for the suggestion! Added the two items.

MaskRay commented:
(release/20.x will be created less one day. Likely this doesn't need more updates. Landing now.)

smithp35 commented:
Apologies had to leave the office before I saw the update. LGTM in post.
